project/packages/dtos/src/questions.ts

```diff
@@ -1,12 +1,62 @@
 import { z } from "zod";
 
+export enum Complexity {
+  Easy = "Easy",
+  Medium = "Medium",
+  Hard = "Hard",
+}
+
+export enum Category {
+  Strings = "Strings",
+  Algorithms = "Algorithms",
+  DataStructures = "Data Structures",
+  BitManipulation = "Bit Manipulation",
+  Recursion = "Recursion",
+  Databases = "Databases",
+  BrainTeaser = "Brain Teaser",
+  Arrays = "Arrays",
+}
+
+const ComplexitySchema = z.nativeEnum(Complexity);
+const CategorySchema = z.nativeEnum(Category);
+
+export const questionSchema = z
+  .object({
+    id: z.bigint(),
+    created_at: z.date(),
+    updated_at: z.date(),
+    deleted_at: z.date().nullable(),
+    q_title: z.string(),
+    q_desc: z.string(),
+    q_category: CategorySchema,
+    q_complexity: ComplexitySchema,
```

Comment:
i think these fields can probably be abstracted into a separate zod object since we are gonna be reusing it in alot of the schemas

Reply:
as in something like 
```
export const questionTitleSchema = z.object({
  q_title: z.string().min(1),
});

export const updateQuestionSchema = z
  .object({
    q_title: questionTitleSchema,
    ...
```
isit

Reply:
LGTM!